cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/GoalViolations.java

```diff
@@ -23,73 +23,67 @@
 public class GoalViolations extends KafkaAnomaly {
   private static final Logger LOG = LoggerFactory.getLogger(GoalViolations.class);
   private final KafkaCruiseControl _kafkaCruiseControl;
-  private final List<Violation> _goalViolations = new ArrayList<>();
+  // The priority order of goals is maintained here.
+  private final List<String> _fixableViolatedGoals;
+  private final List<String> _unfixableViolatedGoals;
   private final boolean _allowCapacityEstimation;
 
   public GoalViolations(KafkaCruiseControl kafkaCruiseControl, boolean allowCapacityEstimation) {
     _kafkaCruiseControl = kafkaCruiseControl;
     _allowCapacityEstimation = allowCapacityEstimation;
+    _fixableViolatedGoals = new ArrayList<>();
+    _unfixableViolatedGoals = new ArrayList<>();
   }
 
   /**
    * Add detected goal violation.
    *
-   * @param priority The priority of the goal.
    * @param goalName The name of the goal.
-   * @param executionProposals Proposals to execute for fixing the detected anomaly. Empty set indicates goal violations
-   *                           due to optimization failures.
+   * @param fixable Whether the violated goal is fixable or not.
    */
-  public void addViolation(int priority, String goalName, Set<ExecutionProposal> executionProposals) {
-    _goalViolations.add(new Violation(priority, goalName, executionProposals));
+  public void addViolation(String goalName, boolean fixable) {
+    if (fixable) {
+      _fixableViolatedGoals.add(goalName);
+    } else {
+      _unfixableViolatedGoals.add(goalName);
+    }
   }
 
   /**
    * Get all the goal violations.
    */
-  public List<Violation> violations() {
-    return _goalViolations;
+  public Set<String> violations() {
```

Comment:
If we use `Map<Boolean, List<String>>` (please see prev. proposal), then would we avoid creating a new `Set` for the response?